### ISSUE TITLE
Please upgrade "go-hclog" library

### ISSUE THREAD

**Issue by onlyjob** [hashicorp/nomad#6810]
Nomad 0.10.2 FTBFS with [github.com/hashicorp/go-hclog](https://github.com/hashicorp/go-hclog) >=0.9.2 as follows:

~~~~
# github.com/hashicorp/nomad/nomad 
src/github.com/hashicorp/nomad/nomad/server.go:1110:29: cannot use logger (type *log.Logger) as type hclog.Logger in assignment: 
        *log.Logger does not implement hclog.Logger (missing Debug method) 
~~~~

Please consider upgrading vendored library to latest [released version](https://github.com/hashicorp/go-hclog/releases).

Thanks.

**Comment by endocrimes** [comment]
Nomad 0.10.2 [vendors go-hclog@234833755cb25ae46996d0ef823326f492f89243](https://github.com/hashicorp/nomad/blob/v0.10.2/vendor/vendor.json#L213), which is the same sha as the latest release [v0.10.0](https://github.com/hashicorp/go-hclog/commit/234833755cb25ae46996d0ef823326f492f89243). :smile:.

It looks like your problem comes from upgrading to a newer version of hashicorp/raft, where it changed from using golang's standard logger to go-hclog in [#360](https://github.com/hashicorp/raft/pull/360).

**Comment by endocrimes** [comment]
It appears that we have not yet tagged a release of hashicorp/raft since the inclusion of that PR however (https://github.com/hashicorp/raft/releases/tag/v1.1.1 is currently the latest)

**Comment by onlyjob** [comment]
I first experienced this problem in 0.10.0 but before submitting this bug report I've reproduced it on current release 0.10.2.

> It looks like your problem comes from upgrading to a newer version of hashicorp/raft

Yes this is precisely _my_ problem (which is actually a downstream problem in Debian where I maintain Nomad packages) but it is also _your_ problem because you have to upgrade the library sooner or later.

Please keep such issues open until the problem is fixed. Denial helps nobody, you know...

With 0.10.1 I used the following patch which no longer applies to 0.10.2:

```
--- a/nomad/server.go
+++ b/nomad/server.go
@@ -1101,10 +1101,15 @@
                s.config.LogOutput)
        s.raftTransport = trans

        // Make sure we set the Logger.
-       logger := s.logger.StandardLogger(&log.StandardLoggerOptions{InferLevels: true})
-       s.config.RaftConfig.Logger = logger
+       raftLogger := log.New(&log.LoggerOptions{
+               Name:       "raft",
+               Output:     s.config.LogOutput,
+               TimeFormat: `2006/01/02 15:04:05`,
+       })
+       s.config.RaftConfig.Logger = raftLogger
+
        s.config.RaftConfig.LogOutput = nil

        // Our version of Raft protocol 2 requires the LocalID to match the network
        // address of the transport. Raft protocol 3 uses permanent ids.
```

**Comment by endocrimes** [comment]
@onlyjob FWIW I was going to upgrade Raft last night before realizing that there is no release with the new logging change yet - I'll make sure we file an issue to do so when a release is cut though.

**Comment by onlyjob** [comment]
OK, thanks. This FTBFS is with raft-1.1.1 and go-hclog-0.10.0.

Could you re-open this issue please?

**Comment by tgross** [comment]
@endocrimes I'm going to reopen so that we can track that change has been merged for the 0.11.0 release.

**Comment by tgross** [comment]
Looks like we needed bugfixes for 0.10.3. Merged in https://github.com/hashicorp/nomad/pull/6872

**Comment by onlyjob** [comment]
Thanks! :)

**Comment by onlyjob** [comment]
Which branch `nomad/server.go` fixes for _go-hclog_ 0.10.2 are committed to? Thanks.

**Comment by tgross** [comment]
Currently just `master`. We'll cut a release branch once we're ready to cut the next release (after the new year).
